### PR TITLE
OfferStrategy and ports in TaskInfo

### PR DESCRIPTION
**Pull request by mwl**
Seems there is a mismatch between the strategy for accepting offer and the actually used ports
* strategy requires port 5000 always to be in the offer;
* however 5000 or 514 or nothing will be used depending on the configuration

### CODE REVIEW

**Comment by mwl** [comment]
I'll be changing collectd port to default `25826/UDP`

File: collectd/logstash.conf [hunk 1]
```diff
@@ -8,7 +8,7 @@ LoadPlugin network
     IgnoreSelected false
 </Plugin>
 <Plugin network>
-    <Server "logstash" "5000">
+    <Server "logstash">
     </Server>
```
Comment (frankscholten) [inline collectd/logstash.conf]: Shouldn't this be port 25826?

File: logstash-executor/src/main/java/org/apache/mesos/logstash/executor/LogstashService.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ private static String serialize(LogstashProtos.LogstashConfiguration logstashCon
...
-                ofConditional(logstashConfiguration.getLogstashPluginInputCollectd(), LogstashProtos.LogstashPluginInputCollectd::isInitialized).map(config -> LS.plugin("udp", LS.map(LS.kv("port", LS.number(5000 /*TODO: config.getPort()*/)), LS.kv("buffer_size", LS.number(1452)), LS.kv("codec", LS.plugin("collectd", LS.map()))))),
+                ofConditional(logstashConfiguration.getLogstashPluginInputCollectd(), LogstashProtos.LogstashPluginInputCollectd::isInitialized).map(config -> LS.plugin("udp", LS.map(LS.kv("port", LS.number(25826 /*TODO: config.getPort()*/)), LS.kv("buffer_size", LS.number(1452)), LS.kv("codec", LS.plugin("collectd", LS.map()))))),
                 ofConditional(logstashConfiguration.getLogstashPluginInputFile(), LogstashProtos.LogstashPluginInputFile::isInitialized).map(config -> LS.plugin("file", LS.map(LS.kv("path", LS.array(config.getPathList().stream().map(path -> "/logstashpaths" + path).map(LS::string).toArray(LS.Value[]::new))))))
```
Comment (frankscholten) [inline logstash-executor/src/main/java/org/apache/mesos/logstash/executor/LogstashService.java]: I see 2 TODOs

File: logstash-scheduler/src/main/java/org/apache/mesos/logstash/scheduler/OfferStrategy.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
     @Inject
     private LogstashConfig logstashConfig;
 
-    private List<Integer> neededPorts = asList(5000); // TODO: 25/11/2015 Configurable
+    private List<Integer> neededPorts = asList(514, 25826); // TODO: 25/11/2015 Configurable
```
Comment (frankscholten) [inline logstash-scheduler/src/main/java/org/apache/mesos/logstash/scheduler/OfferStrategy.java]: Put them in LogstashConstants

File: logstash-scheduler/src/main/java/org/apache/mesos/logstash/scheduler/TaskInfoBuilder.java [hunk 4]
```diff
@@ -39,7 +42,7 @@
             dockerExecutor.addPortMappings(Protos.ContainerInfo.DockerInfo.PortMapping.newBuilder().setHostPort(514).setContainerPort(514).setProtocol("udp"));
         }
         if (features.isCollectd()) {
-            dockerExecutor.addPortMappings(Protos.ContainerInfo.DockerInfo.PortMapping.newBuilder().setHostPort(5000).setContainerPort(5000).setProtocol("udp"));
+            dockerExecutor.addPortMappings(Protos.ContainerInfo.DockerInfo.PortMapping.newBuilder().setHostPort(25826).setContainerPort(25826).setProtocol("udp"));
         }
```
Comment (frankscholten) [inline logstash-scheduler/src/main/java/org/apache/mesos/logstash/scheduler/TaskInfoBuilder.java]: Also LogstashConstants

Comment (mwl) [inline logstash-scheduler/src/main/java/org/apache/mesos/logstash/scheduler/TaskInfoBuilder.java]: The Constants are dead ;)
Configuration is the future. I'll do a S/R on those ports.

Comment (frankscholten) [inline logstash-scheduler/src/main/java/org/apache/mesos/logstash/scheduler/TaskInfoBuilder.java]: Ok fine.

File: collectd/logstash.conf (hunk 1 above)
Comment (mwl) [inline collectd/logstash.conf]: Yup, that's default

> The optional second argument specifies a port number or a service name. If not given, the default, 25826, is used.

https://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_network

Comment (mwl) [inline collectd/logstash.conf]: Bare in mind. It's only used for demo

**Pull request merged**